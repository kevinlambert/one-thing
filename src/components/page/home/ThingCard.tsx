import { useNavigate } from "react-router-dom";
import { View, Text, Divider, Button, useTheme } from "@aws-amplify/ui-react";
import LinkIcon from "./LinkIcon";

type ToDoProps = {
  path: string;
};

const ToDo = ({ path }: ToDoProps) => {
  const navigate = useNavigate();

  const onClickHandler = () => {
    navigate(path);
  };

  return (
    <View>
      <Button size="small" variation="primary" onClick={onClickHandler}>
        Set your ONE Thing
      </Button>
    </View>
  );
};

type Props = {
  periodTitle: string;
  thingContent?: string;
  isLinked?: boolean;
  isLinkedIcon?: boolean;
  showLink?: boolean;
  borderTopRound?: boolean;
  borderBottomRound?: boolean;
  toDoPath: string;
};

export default ({
  periodTitle,
  thingContent,
  isLinked = false,
  isLinkedIcon = false,
  showLink = false,
  borderTopRound = false,
  borderBottomRound = false,
  toDoPath,
}: Props) => {
  const { tokens } = useTheme();

  return (
    <View
      position={"relative"}
      padding={tokens.space.medium}
      borderRadius={
        borderTopRound
          ? `${tokens.space.medium} ${tokens.space.medium} 0 0`
          : borderBottomRound
          ? `0 0 ${tokens.space.medium} ${tokens.space.medium}`
          : 0
      }
      borderWidth={tokens.borderWidths.small}
      borderStyle={"solid"}
      borderColor={
        isLinked ? tokens.colors.green[60] : tokens.colors.neutral[40]
      }
      backgroundColor={
        isLinked ? tokens.colors.green[10] : tokens.colors.neutral[10]
      }
    >
      {showLink && <LinkIcon isLinked={isLinkedIcon} />}
      <Text marginBottom={tokens.space.xxs} fontSize={tokens.fontSizes.xs}>
        {periodTitle}
      </Text>
      <Divider size="small" marginBottom={tokens.space.xxs} />
      <Text fontSize={tokens.fontSizes.large}>{thingContent}</Text>
      {!thingContent && <ToDo path={toDoPath} />}
    </View>
  );
};
