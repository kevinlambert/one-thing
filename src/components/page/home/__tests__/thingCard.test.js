import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import ThingCard from "../ThingCard";

const TITLE = "TITLE";
const CONTENT = "CONTENT";
const IS_LINKED = true;
const TODO_PATH = "TODO_PATH";

it("renders the content", () => {
  const div = document.createElement("div");
  render(
    <MemoryRouter>
      <ThingCard
        periodTitle={TITLE}
        thingContent={CONTENT}
        isLinked={IS_LINKED}
        toDoPath={TODO_PATH}
      />
    </MemoryRouter>,
    div
  );

  expect(screen.getByText(TITLE)).toBeInTheDocument();
  expect(screen.getByText(CONTENT)).toBeInTheDocument();
});

it("renders the set thing button", () => {
  const div = document.createElement("div");
  render(
    <MemoryRouter>
      <ThingCard toDoPath={TODO_PATH} />
    </MemoryRouter>,
    div
  );

  expect(screen.getByText("Set your ONE Thing")).toBeInTheDocument();
});

it("snapshot - not linked", () => {
  const div = document.createElement("div");
  const tree = renderer
    .create(
      <MemoryRouter>
        <ThingCard
          periodTitle={TITLE}
          thingContent={CONTENT}
          toDoPath={TODO_PATH}
        />
      </MemoryRouter>,
      div
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("snapshot - linked", () => {
  const div = document.createElement("div");
  const tree = renderer
    .create(
      <MemoryRouter>
        <ThingCard
          periodTitle={TITLE}
          thingContent={CONTENT}
          isLinked={true}
          toDoPath={TODO_PATH}
        />
      </MemoryRouter>,
      div
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("snapshot - linked icon", () => {
  const div = document.createElement("div");
  const tree = renderer
    .create(
      <MemoryRouter>
        <ThingCard
          periodTitle={TITLE}
          thingContent={CONTENT}
          isLinkedIcon
          showLink
          toDoPath={TODO_PATH}
        />
      </MemoryRouter>,
      div
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("snapshot - rounded corners top", () => {
  const div = document.createElement("div");
  const tree = renderer
    .create(
      <MemoryRouter>
        <ThingCard borderTopRound />
      </MemoryRouter>,
      div
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("snapshot - rounded corners bottom", () => {
  const div = document.createElement("div");
  const tree = renderer
    .create(
      <MemoryRouter>
        <ThingCard borderBottomRound />
      </MemoryRouter>,
      div
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
