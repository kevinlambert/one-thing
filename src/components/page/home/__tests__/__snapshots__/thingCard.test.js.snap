// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot - linked 1`] = `
<div
  inert={null}
  style={
    Object {
      "backgroundColor": "var(--amplify-colors-green-10)",
      "borderColor": "var(--amplify-colors-green-60)",
      "borderStyle": "solid",
      "borderWidth": "var(--amplify-border-widths-small)",
      "padding": "var(--amplify-space-medium)",
      "position": "relative",
    }
  }
>
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-xs)",
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  >
    TITLE
  </p>
  <hr
    aria-orientation="horizontal"
    className="amplify-divider amplify-divider--horizontal amplify-divider--small"
    data-size="small"
    inert={null}
    style={
      Object {
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  />
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-large)",
      }
    }
  >
    CONTENT
  </p>
</div>
`;

exports[`snapshot - linked icon 1`] = `
<div
  inert={null}
  style={
    Object {
      "backgroundColor": "var(--amplify-colors-neutral-10)",
      "borderColor": "var(--amplify-colors-neutral-40)",
      "borderStyle": "solid",
      "borderWidth": "var(--amplify-border-widths-small)",
      "padding": "var(--amplify-space-medium)",
      "position": "relative",
    }
  }
>
  <div
    className="amplify-flex linkIcon"
    inert={null}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "var(--amplify-colors-white)",
        "borderColor": "var(--amplify-colors-green-60)",
        "borderRadius": "16px",
        "borderStyle": "solid",
        "borderWidth": "var(--amplify-border-widths-small)",
        "color": "var(--amplify-colors-green-60)",
        "height": "32px",
        "justifyContent": "center",
        "position": "absolute",
        "top": "-16px",
        "width": "32px",
      }
    }
  >
    <svg
      className="amplify-icon"
      fill="currentColor"
      height="1em"
      inert={null}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "height": "22px",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
      />
    </svg>
  </div>
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-xs)",
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  >
    TITLE
  </p>
  <hr
    aria-orientation="horizontal"
    className="amplify-divider amplify-divider--horizontal amplify-divider--small"
    data-size="small"
    inert={null}
    style={
      Object {
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  />
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-large)",
      }
    }
  >
    CONTENT
  </p>
</div>
`;

exports[`snapshot - not linked 1`] = `
<div
  inert={null}
  style={
    Object {
      "backgroundColor": "var(--amplify-colors-neutral-10)",
      "borderColor": "var(--amplify-colors-neutral-40)",
      "borderStyle": "solid",
      "borderWidth": "var(--amplify-border-widths-small)",
      "padding": "var(--amplify-space-medium)",
      "position": "relative",
    }
  }
>
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-xs)",
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  >
    TITLE
  </p>
  <hr
    aria-orientation="horizontal"
    className="amplify-divider amplify-divider--horizontal amplify-divider--small"
    data-size="small"
    inert={null}
    style={
      Object {
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  />
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-large)",
      }
    }
  >
    CONTENT
  </p>
</div>
`;

exports[`snapshot - rounded corners bottom 1`] = `
<div
  inert={null}
  style={
    Object {
      "backgroundColor": "var(--amplify-colors-neutral-10)",
      "borderColor": "var(--amplify-colors-neutral-40)",
      "borderRadius": "0 0 var(--amplify-space-medium) var(--amplify-space-medium)",
      "borderStyle": "solid",
      "borderWidth": "var(--amplify-border-widths-small)",
      "padding": "var(--amplify-space-medium)",
      "position": "relative",
    }
  }
>
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-xs)",
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  />
  <hr
    aria-orientation="horizontal"
    className="amplify-divider amplify-divider--horizontal amplify-divider--small"
    data-size="small"
    inert={null}
    style={
      Object {
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  />
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-large)",
      }
    }
  />
  <div
    inert={null}
    style={Object {}}
  >
    <button
      className="amplify-button amplify-field-group__control amplify-button--primary amplify-button--small"
      data-fullwidth={false}
      data-size="small"
      data-variation="primary"
      inert={null}
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      Set your ONE Thing
    </button>
  </div>
</div>
`;

exports[`snapshot - rounded corners top 1`] = `
<div
  inert={null}
  style={
    Object {
      "backgroundColor": "var(--amplify-colors-neutral-10)",
      "borderColor": "var(--amplify-colors-neutral-40)",
      "borderRadius": "var(--amplify-space-medium) var(--amplify-space-medium) 0 0",
      "borderStyle": "solid",
      "borderWidth": "var(--amplify-border-widths-small)",
      "padding": "var(--amplify-space-medium)",
      "position": "relative",
    }
  }
>
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-xs)",
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  />
  <hr
    aria-orientation="horizontal"
    className="amplify-divider amplify-divider--horizontal amplify-divider--small"
    data-size="small"
    inert={null}
    style={
      Object {
        "marginBlockEnd": "var(--amplify-space-xxs)",
      }
    }
  />
  <p
    className="amplify-text"
    inert={null}
    style={
      Object {
        "fontSize": "var(--amplify-font-sizes-large)",
      }
    }
  />
  <div
    inert={null}
    style={Object {}}
  >
    <button
      className="amplify-button amplify-field-group__control amplify-button--primary amplify-button--small"
      data-fullwidth={false}
      data-size="small"
      data-variation="primary"
      inert={null}
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      Set your ONE Thing
    </button>
  </div>
</div>
`;
